patches/server/0807-Add-PlayerShieldDisableEvent.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 20 Sep 2021 13:27:50 -0500
+Subject: [PATCH] Add PlayerShieldDisableEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index 8a864238e154e2131834d013652746b7e7a78c97..91afdf9b0bfd44e2dbfd1c36cbd35b50a6fbb407 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -1588,9 +1588,16 @@ public abstract class Mob extends LivingEntity {
+             float f = 0.25F + (float) EnchantmentHelper.getBlockEfficiency(this) * 0.05F;
+ 
+             if (this.random.nextFloat() < f) {
++                // Paper start
++                io.papermc.paper.event.player.PlayerShieldDisableEvent event = new io.papermc.paper.event.player.PlayerShieldDisableEvent((org.bukkit.entity.Player) player.getBukkitEntity(), getBukkitEntity());
++                event.callEvent();
++
++                if (!event.isCancelled()) {
```

Comment:
You don't need a reference to the field, so this can be simplified even further than the other comment. Just construct it, call it, and check if its cancelled by looking at the return value of `Event#callEvent`. Should only be a 1 line diff with a single paper comment at the end.